sdk/eventhub/Azure.Messaging.EventHubs/src/Consumer/EventHubConsumerClient.cs

```diff
@@ -538,15 +537,17 @@ await foreach (var partitionEvent in eventChannel.Reader.EnumerateChannel(option
                 // Determine the partitions for the Event Hub and create the shared channel.
 
                 var partitions = await GetPartitionIdsAsync(cancellationToken).ConfigureAwait(false);
-                eventChannel = CreateEventChannel((BackgroundPublishReceiveBatchSize * partitions.Length * 2));
+
+                var channelSize = options.CacheEventCount * partitions.Length * 2L;
```

Comment:
This method is not performance-critical and we advise not calling it for production scenarios.  The potential for having a smaller channel size than is ideal for very large values is not a concern.